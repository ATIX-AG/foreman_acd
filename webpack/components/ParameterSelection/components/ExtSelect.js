import React from 'react';
import PropTypes from 'prop-types';
import Select from 'foremanReact/components/common/forms/Select';

const ExtSelect= ({
  hidden,
  editable,
  viewText,
  selectValue,
  onChange,
  options,
  additionalData,
}) =>{
  if (hidden) {
    return null;
  }

  if (!editable) {
    return (
     <div>{viewText}</div>
    );
  }

  return (
    <Select
       value={selectValue}
       onChange={e => onChange(e.target.value, additionalData) }
       options={options}
    />
  );
};

ExtSelect.propTypes = {
  hidden: PropTypes.bool.isRequired,
  editable: PropTypes.bool.isRequired,
  viewText: PropTypes.string,
  selectValue: PropTypes.string,
  onChange: PropTypes.func.isRequired,
  options: PropTypes.object,
  additionalData: PropTypes.object,
};

export default ExtSelect;
