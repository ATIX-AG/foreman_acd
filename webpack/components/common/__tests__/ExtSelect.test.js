import { testComponentSnapshotsWithFixtures } from 'react-redux-test-utils';

import ExtSelect from '../ExtSelect';

const noop = () => {};

const fixtures = {
  'should render selection field': {
    hidden: false,
    editable: false,
    viewText: '',
    selectValue: '1',
    onChange: noop,
    options: { first: 'first', second: 'second' },
    additionalData: { moreData: 'moooore' },
  },
  'should render hidden selection field': {
    hidden: true,
    editable: false,
    viewText: '',
    selectValue: null,
    onChange: noop,
    options: {},
    additionalData: {},
  },
  'should render disabled button': {
    hidden: false,
    editable: true,
    viewText: 'Awesome Text',
    selectValue: '0',
    onChange: noop,
    options: { first: 'first', second: 'second' },
    additionalData: {},
  },
};

describe('ExtSelect', () =>
  testComponentSnapshotsWithFixtures(ExtSelect, fixtures));
