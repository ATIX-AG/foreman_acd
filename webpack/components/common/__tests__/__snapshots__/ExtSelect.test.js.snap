// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExtSelect should render disabled button 1`] = `
<Select
  onChange={[Function]}
  options={
    Object {
      "first": "first",
      "second": "second",
    }
  }
  value="0"
/>
`;

exports[`ExtSelect should render hidden selection field 1`] = `""`;

exports[`ExtSelect should render selection field 1`] = `<div />`;
